workers/unity/Assets/Playground/Editor/Prefabs/PrefabPreprocessor.cs

```diff
@@ -36,39 +36,13 @@ private static void PlaymodeStateChanged(PlayModeStateChange playModeStateChange
         }
 
         private static void PreprocessPrefabs()
-        {
-            if (!DoPrefabsNeedFixing())
-            {
-                return;
-            }
-
-            FixPrefabs();
-
-            AssetDatabase.SaveAssets();
-        }
-
-        private static bool DoPrefabsNeedFixing()
-        {
-            return AssetDatabase.FindAssets("t:Prefab")
-                .Select(AssetDatabase.GUIDToAssetPath)
-                .Select(AssetDatabase.LoadAssetAtPath<GameObject>)
-                .Any(PrefabNeedsFixing);
-        }
-
-        private static bool PrefabNeedsFixing(GameObject prefabObject)
-        {
-            return prefabObject != null
-                && prefabObject.GetComponents<MonoBehaviour>()
-                    .Any(BehaviourNeedsFixing);
-        }
-
-        private static void FixPrefabs()
         {
             var prefabsToFix = new List<GameObject>();
```

Comment:
Can this be made a `HashSet<GameObject>` that you just always add the prefab to if a monobehaviour on it needs to be disabled?
This saves some statekeeping and will ensure it is not duplicated.

Reply:
Removed the list, marking it dirty as it's being processed now